### DISCUSSION TITLE
Code Fold Indicators - Show horizontal lines at each code fold

### DISCUSSION THREAD

**Discussion by advice2020** [geany/geany#3987] [Ideas]
Suggesting to add an option that would allow a thin horizontal line across screen at each code fold that would always be visible. These would act as visual separators between code folds.

Thin horizontal lines like this are already shown when code folds are collapsed. So in the following link
https://github.com/geany/geany/discussions/3998
I made a suggestion for collapsed code fold horizontal lines to be shown as a bold/thick line instead.

Thank You

**Comment by elextr** [comment]
See response to #3986 try setting `folding_horiz_line`.

**Comment by advice2020** [comment]
Thank you for responding, `folding_horiz_line` is for showing horizontal lines when code folds are collapsed, this is for showing horizontal lines at code folds when expanded.
Probably not interested, but I have updated my original post above to hopefully be a little clearer.
Thanks again

**Comment by elextr** [comment]
There are a number of options for the line(s) in Scintilla [here](https://www.scintilla.org/ScintillaDoc.html#SCI_SETFOLDFLAGS).

However whoever whenever they wrote it limited the values accepted from `folding_horiz_line` for some reason and not the whole range.

"Somebody" could make a PR that added a new setting that could accept the whole range and override `folding_horiz_line` if it was set.

**Comment by advice2020** [comment]
@nyamatongwe
Hey Neil, I have some posts regarding code fold visual indicators that I have been told are probably Scintilla related.

I have others but these two might not get approval so I figured I would start with them.  I was wondering when you have a minute, if you can take a look at this post as well as the one linked 3998 to it to answer the following questions.

I made a little copy and paste template below to hopefully make things a little easier for you to answer.

3987 - Horizontal lines at code folds
-Is Scintilla required for this = yes/no
-If so would this be something considered to be added to Scintilla = yes/no

3998 - Bold collapsed horizontal lines
-Is Scintilla required for this = yes/no
-If so would this be something considered to be added to Scintilla = yes/no

Thank You

**Reply by nyamatongwe** [reply]
Some of this is answered by the documentation referenced by @elextr above. You should experiment with the fold flags API.

**Comment by advice2020** [comment]
@nyamatongwe
Hey Neil, thank you for your response.
Little confused but I will try to ask Geany users.

Apologize, hopefully I am not bothering you.
Please bare with me, only a few more posts to verify if you are interested.
https://github.com/geany/geany/issues/4427
https://github.com/geany/geany/discussions/4005

The only other post I know of is another one about code folds but having a tough time creating that post at the moment.
Thanks again for your help

**Reply by nyamatongwe** [reply]
There is a Scintilla mailing list at https://groups.google.com/forum/#!forum/scintilla-interest . Possibly some other community members will be more interested in engaging with your issues than I am.
